src/Libraries/CoreNodeModels/DropDown.cs

```diff
@@ -63,13 +63,24 @@ public ObservableCollection<DynamoDropDownItem> Items
                 RaisePropertyChanged("Items");
             }
         }
+
         public override NodeInputData InputData
         {
             //TODO There is not yet an appropriate input type
             //defined in the cogs graph schema to support dropdowns
             //which return arbitrary objects at some index - implement this
             //when that exists.
-            get { return null; }
+            get
+            {
+                return new NodeInputData
+                {
+                    Id = this.GUID,
+                    Name = this.Name,
+                    Type = NodeInputTypes.itemSelection,
+                    Description = this.Description,
+                    Value = this.SelectedString
```

Comment:
shouldn't this also serialize the `SelectedIndex`? @QilongTang I know you ran into trouble and needed both index and selected value when deserializing dropdowns - is that correct?

Reply:
Yes, I would also recommend saving index for additional safe guard. This is because there is no guarantee that in the new context the list options will be same or the serialized selection is still available.

Reply:
@mjkkirschner @QilongTang - Would you propose that we add a new `SelectedIndex` to the [`NodeInputData` model](https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Graph/Nodes/NodeInputData.cs)?

```csharp
        /// <summary>
        /// The index of the selected index.
        /// </summary>
        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
        public int SelectedIndex { get; set; }

Reply:
Yes you can refer to https://github.com/DynamoDS/Dynamo/blob/master/src/Libraries/CoreNodeModels/DropDown.cs#L80 to see what kind of problems we are guarding